### PR TITLE
#31 Create reusable text and image banner (side by side)

### LINKED ISSUE #31: 🎨 Create reusable text and image banner (side by side)
#  Create reusable text and image banner (side by side)
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a component I can reuse throughout the site to display images and text side by side.

### Description

This component displays text and images side by side on the site

Then section has the following as inputs to the component:

```
1. An image url
2. A paragraph of text
3. A title
4. Placement of image ("left" or "right")
5. Background color of the section (Hex Code i.e "#00000")

```

Screenshot 2023-02-06 at 10 24 37

Screenshot 2023-02-06 at 10 24 53

Your assignment is to implement the component to appear as above.

Please take note of all the design features (including mobile and rounded corners)

#### Link to full design:
```
https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/31438bb0-400a-4da8-a374-c0440535c37c

mobile 
-

```
### Technical Analysis

This component will be created in libs/features/components/banners

The component will be called elewa-group-image-and-text-banner

https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690301/elewa-group-website/Images/gettyimages-525701055-2048x2048_g7nbt1.png

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 50 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
4. The code is isolated to the libs/page/home package, unless a justified reason is specified.
5. The component is aligned using CSS flex
6. The design is responsive for screen, tablet and phone
7. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by lilmithi**
# Description
This has been built in collaboration with @royokite 

- We have designed a reusable text and image banner in which the image and text render according to the side specified with the default set to 'left'(image on the left, text on the right')
- The component accepts a title, text, image url, background color and placement('left' or 'right')

Fixes #31  

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# Screenshot
White background (Image left) Desktop
leftwhitedesktop

White background (Image right) Desktop
rightwhitedesktop

White background (Image right) Mobile
rightwhitemobile

White background (Image left) Mobile
leftwhitemobile

Black background (Image left) Desktop
leftblackdesktop

Black background (Image right) Desktop
rightblackdesktop

Black background (Image right) Mobile
rightblackmobile

Black background (Image left) Mobile
leftblackmobile

# How Has This Been Tested?

- [x] Tested with different browsers for any breaking changes

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by lilmithi** [comment]
@IanOdhiambo9 awaiting your review

File: libs/elements/layout/src/lib/components/banners/elewa-group-image-and-text-banner/elewa-group-image-and-text-banner.component.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<div
+  class="banner-container"
+  *ngIf="placement !== 'right'"
+  [ngStyle]="{ 'background-color': background }"
+>
```
Comment (IanOdhiambo9) [inline libs/elements/layout/src/lib/components/banners/elewa-group-image-and-text-banner/elewa-group-image-and-text-banner.component.html:5]: use *ngIf="placement == 'right'; else  'imageLeft'"

and put and #imageLeft tag on an ng-template

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]
Good Job on this but the component was put in the wrong place. Please move to  libs/features/components/banners as instructed on the issue template

**Comment by IanOdhiambo9** [comment]
Do PR to staging branch

**Pull request closed without merging**